twine-upload.sh

```diff
@@ -40,6 +40,12 @@ INPUT_VERIFY_METADATA="$(get-normalized-input 'verify-metadata')"
 INPUT_SKIP_EXISTING="$(get-normalized-input 'skip-existing')"
 INPUT_PRINT_HASH="$(get-normalized-input 'print-hash')"
 
+if [[ "${INPUT_USER}" == "__token__" && -z "${INPUT_PASSWORD}" ]] ; then
```

Comment:
@woodruffw do you think this approach https://github.com/orgs/community/discussions/25259#discussioncomment-3247158 could detect whether `${GITHUB_TOKEN}` has the correct `id-token: write` privilege at this point and skip going into the OIDC flow if it doesn't?

Reply:
That _might_ work, although I'm not actually sure `id-token: write` corresponds to a documented OAuth scope: https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps

Given that, we might end up depending on a scope that GitHub could change at any time (as an implementation detail of how their ephemeral OIDC endpoint works).

Reply:
Fair. Let's not depend on undocumented things, then. Resolving.